skbio/io/format/stockholm.py

```diff
@@ -0,0 +1,322 @@
+'RA': '   Deiman BA, Kortlever RM, Pleij CW;'
+'RL': '   J Virol 1997;71:5990-5996.'
+Positional metadata:
+'SS_cons': <dtype: object>
+Stats:
+sequence count: 4
+position count: 23
+-------------------------------------------------
+UGAGUUCUCGAUCUCUAAAAUCG
+UGAGUUCUCUAUCUCUAAAAUCG
+UAAGUUCUCGAUCUUUAAAAUCG
+UAAGUUCUCGAUCUCUAAAAUCG
+
+References
+==========
+.. [1] https://en.wikipedia.org/wiki/Stockholm_format
+.. [2] http://sonnhammer.sbc.su.se/Stockholm.html
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from collections import OrderedDict, namedtuple
+
+from skbio.alignment import TabularMSA
+from skbio.sequence import Protein
+from skbio.io import create_format, StockholmFormatError
+
+stockholm = create_format('stockholm')
+_SeqData = namedtuple("SeqData", ["seq", "metadata", "pos_metadata"])
+
+
+@stockholm.sniffer()
+def _stockholm_sniffer(fh):
+    # Smells a Stockholm file if the first line contains 'Stockholm' and the
+    # version number.
+    try:
+        line = next(fh)
+    except StopIteration:
+        return False, {}
+    if line[:15] == "# STOCKHOLM 1.0":
+        return True, {}
+
+    return False, {}
+
+
+@stockholm.reader(TabularMSA)
+def _stockholm_to_tabular_msa(fh, constructor=Protein):
+    # Uses OrderedDict() to make sure dna_data isn't arranged randomly
+    dna_data = OrderedDict()
+    # Creates SeqData class to store all sequence data and make code more
+    # readable
+    metadata = {}
+    positional_metadata = {}
+    seqs = []
+
+    # Retrieves data from file, reads first so that data order will be kept
+    # consistent.
+    for line in fh:
+        if is_data_line(line):
+            dna_data = _parse_stockholm_line_data(line, dna_data)
+    fh.seek(0)
+
+    # Retrieves metadata from file
+    for line in fh:
+        if line.startswith("#=GF"):
+            metadata = _parse_stockholm_line_gf(line, metadata)
+        elif line.startswith("#=GS"):
+            dna_data = _parse_stockholm_line_gs(line, dna_data)
+        elif line.startswith("#=GR"):
+            dna_data = _parse_stockholm_line_gr(line, dna_data)
+        elif line.startswith('#=GC'):
+            positional_metadata = _parse_stockholm_line_gc(line,
+                                                           positional_metadata)
+
+    for key in dna_data.keys():
+        # Sets blank dictionaries and lists to None instead
+        # Note: _replace is not a private function, see
+        # https://docs.python.org/2/library/collections.html#namedtuple-factory
+        # -function-for-tuples-with-named-fields
```

Comment:
Useful link! Maybe link directly to the subsection that mentions the underscores though:
https://docs.python.org/2/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields